### PR TITLE
Add troubleshooting page with PCI Bus ID issue description

### PR DESCRIPTION
**Pull request by pentschev**

### CODE REVIEW

**Review by madsbk — APPROVED** [review]

**Comment by pentschev** [comment]
Will wait until EOD for any comments and then merge it. Thanks @madsbk for the approval!

File: docs/source/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Troubleshooting
+===============
+
+This is a list of common issues encountered with Dask-CUDA and various systems.
+
+Wrong Device Indexing
+---------------------
+
+It's common to rely on the device indexing presented by ``nvidia-smi`` when creating workers, and that is the default in Dask-CUDA. In most cases, ``nvidia-smi`` provides a one-to-one mapping with ``CUDA_VISIBLE_DEVICES``, but in some systems the ordering may not match. While, ``CUDA_VISIBLE_DEVICES`` indexes GPUs by their PCI Bus ID, ``nvidia-smi`` orders by fastest GPUs. Issues are commonly seen in `DGX Station A100 <https://www.nvidia.com/en-us/data-center/dgx-station-a100/>`_ that contains 4 A100 GPUs, plus a Display GPU, but the Display GPU may not be the last GPU according to the PCI Bus ID. To correct that and ensure the mapping according to the PCI Bus ID, it's necessary to set the ``CUDA_DEVICE_ORDER=PCI_BUS_ID`` environment variable when starting the Python process:
```
Comment (mmccarty) [inline docs/source/troubleshooting.rst:9]: Can you line wrap this paragraph?

**Review by mmccarty — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

**Comment by jakirkham** [comment]
Thanks Peter for the PR & detective work (thank Mike for helping with this)! Also thanks Mads and Mike for the reviews! 😄

**Comment by pentschev** [comment]
Thanks to everyone mentioned by @jakirkham already, and especially thanks to @ayushdg for pointing to the actual solution with `CUDA_DEVICE_ORDER=PCI_BUS_ID`!

**Pull request merged**